hazelcast-client/src/main/java/com/hazelcast/client/cache/impl/NearCachedClientCacheProxy.java

```diff
@@ -242,28 +241,39 @@ protected void getAllInternal(Set<? extends K> keys, Collection<Data> dataKeys,
             return;
         }
 
+        Map<Object, Long> reservations = createHashMap(ncKeys.size());
+        Map<Data, Object> reverseKeyMap = null;
```

Comment:
Instead we have a reverse lookup map, which is just populate if `serializeKeys` is `false`.